javascript-challenges.test.js

```diff
@@ -24,10 +31,64 @@ const secretCodeWord3 = "Eccentric"
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+const codedMessage = (string) => {
+    // create a conditional statement that goes through each word and replaces the letters with numbers
+    if (string.includes("a") || string.includes("A")){
+    return string.replace(/a/gi, "4")
+    } else if (string.includes("e") || string.includes("E")){
+    return string.replace(/e/gi, "3")
+    } else if (string.includes("i") || string.includes("I")){
+        return string.replace(/i/gi, "1")
+    } else if (string.includes("o") || string.includes("O")){
+        return string.replace(/o/gi, "0")
+    } else {
+        return string
+    }
+}
+// the test failed on the first variable. I tried using .map and .filter instead but couldnt figure out how to use it on every string. I ended up just leaving the function like this.
+// yarn run v1.22.19
+// warning package.json: No license field
+// $ /Users/learnacademy/Desktop/week-5-assessment-Danny-DeLaRosa/node_modules/.bin/jest
+//  FAIL  ./javascript-challenges.test.js
+//   codedMessage
+//     ✕ takes in a string and returns a string with a coded message. (2 ms)
 
+//   ● codedMessage › takes in a string and returns a string with a coded message.
+
+//     expect(received).toEqual(expected) // deep equality
+
+//     Expected: "L4ck4d41s1c4l"
+//     Received: "L4ck4d4isic4l"
+
+//       16 | describe ("codedMessage", () => {
+//       17 |     it ("takes in a string and returns a string with a coded message.", () => {
+//     > 18 |         expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+//          |                                               ^
+//       19 |         expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//       20 |         expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+//       21 |     })
+
+//       at Object.toEqual (javascript-challenges.test.js:18:47)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.116 s, estimated 1 s
+// Ran all test suites.
+// error Command failed with exit code 1.
+// info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
+
+// console.log(codedMessage(secretCodeWord1)
```

Comment:
So close!  Remember that .map is an array method so you have to convert the string to and array first. 
Keep working on JS, Danny.  I'm happy to meet with your or connect you with a mentor/tutor who can help, if you'd like.  Just let me know.  Hang in there, you are smart and you will get this.